### PR TITLE
Inventa update 1.1.0

### PR DESCRIPTION
**Pull request by andriidomasov**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
None

## Description
Updated version of Inventa Integration features migration to a new API since the previously used DSAR-API will no longer be updated by Inventa. Thus, in all further updates only the PII-API will be used. Update includes some additional fields, simplified workflow and less restricted constraints.

## Screenshots
DSAR_Inventa_Handler_Sun_Jun_05_2022

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [x] Yes
       - Further details: some of previously used commands, tightly related to DSAR_API, will not execute. None of them, however, are used in new version of playbook.
   - [ ] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Review by idovandijk — COMMENTED** [review]
Thank you for the contribution. I've reviewed the playbooks and got some questions and comments:

1. What does the `inventa-validate-incident-inputs` command do? If it just checks whether the data exists then please use a conditional task for that instead of a script.
Also, what happens if the validation fails? Is there an error? Do we want the incident to stay open with the error or do we want some form of handling for that scenario?
2. Do the keys specified in the `SetGridField` always exist? Or are they different for each customer of Inventa? If different, then that task would fail for other users 
image
3. Please verify that you have a value in `inventadsardatasubjectid` before searching for incidents with it. Otherwise I believe that it may find all recent incidents in XSOAR which may lead to undesired results.
4. Regarding the fields, we're trying to avoid adding pack-specific fields that could be common fields and could be useful for others packs to use. I'll double check this internally to see whether those are an issue or not.

I would like to have a short demo to see how the incidents look like (running playbook, layout), can you give me some possible time windows so we can schedule? Should be 20min or less :)

**Comment by idovandijk** [comment]
> Hi @andriidomasov, thanks for the answers. Regarding the fields, the approach should be:
- If those fields already exist in the Common Pack - use them.
- If those fields could/may be used in other packs in the future - add them to Common Fields.
- If those fields are specific to the pack/product - keep them in their own pack.

From my observation, at least some fields can be added to the CommonTypes pack. For example "Full Name" is a field that could be used in other packs - as any pack or integration that may refer to users or employees who have the "Full Name" property could benefit from that. Therefore, the Full Name field should be a Common field. Please review the fields and decide accordingly for each one, and then make the changes accordingly in your pack where you reference those fields.

As for the demo, seems like we are in the same timezone (Jerusalem and Kyiv time). So, could one of these time windows work? We would need approximately 20-30 minutes. I could do 30 minutes during any of these times this week:
June 14th: 16:30-19:00
June 15th: 14:30-15:30 / 17:00-19:00
June 16th: 14:30-19:00
After the demo I will have a better understanding of the rest of the points.
Please let me know when would be best for you.

**Comment by andriidomasov** [comment]
Thank you for your clarification in regards to the Common Fields. Will review the fields we use and amend them accordingly. As far as I understand, fields added to the CommonTypes pack shouldn't be marked as 'Inventa' in their respective names.
As for the demo, June 14th / 17:00 suits perfectly for me.

**Comment by idovandijk** [comment]
> Thank you for your clarification in regards to the Common Fields. Will review the fields we use and amend them accordingly. As far as I understand, fields added to the CommonTypes pack shouldn't be marked as 'Inventa' in their respective names. As for the demo, June 14th / 17:00 suits perfectly for me.

Correct, common fields should not have the pack name in them.
I will send out an invite. Thank you.

**Comment by andriidomasov** [comment]
Please also add my superior pavlo.rusyn@1touch.io to this meeting. Though it's unclear whether he will attend or not since he's on holiday right now, he promised he would do his best to connect as well. Thanks.

**Comment by idovandijk** [comment]
> Please also add my superior [pavlo.rusyn@1touch.io](mailto:pavlo.rusyn@1touch.io) to this meeting. Though it's unclear whether he will attend or not since he's on holiday right now, he promised he would do his best to connect as well. Thanks.

Done, note that it was auto-declined, probably due to what you said.

**Review by ilappe — CHANGES_REQUESTED** [review]
Thanks for contributing!
please see few small comments to fix
thanks!

File: Packs/Inventa/Integrations/Inventa/Inventa_test.py [hunk 1]
```diff
@@ -211,7 +222,16 @@ def test_get_datasubjects_cmd(mocker):
     main()
     assert demisto.results.call_count == 1
     results = demisto.results.call_args[0]
-    assert results[0]["Contents"] == mock_data.get(command_name, "")
+    # assert results[0]["Contents"] == mock_data.get(command_name, "")
```
[the remark this answers is not in the document]
Comment (andriidomasov) [reply Packs/Inventa/Integrations/Inventa/Inventa_test.py]: removed comment

File: Packs/Inventa/Integrations/Inventa/Inventa_test.py [hunk 2]
```diff
@@ -134,6 +143,8 @@ def mocker_automate(mocker, command_name, method_names):
     mocker.patch.object(demisto, "args", return_value=mock_args(command_name))
     mocker.patch.object(demisto, "command", return_value=command_name)
     mocker.patch.object(demisto, 'results')
+    # mocker.patch("demistomock.callingContext", return_value={}, new_callable=mocker.PropertyMock)
+    # mocker.patch.object(demisto, "callingContext", return_value={})
```
[the remark this answers is not in the document]
Comment (andriidomasov) [reply Packs/Inventa/Integrations/Inventa/Inventa_test.py]: removed comment

File: Packs/Inventa/ReleaseNotes/1_1_0.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+- **Inventa Vehicle Number**
+- **Inventa Passport Number**
+
+#### Incident Types
+- **iDSAR**
+
+#### Integrations
+##### 1Touch.io's Inventa Connector
+- Updated the Docker image to: *demisto/python3:3.10.4.29342*.
+
```
[the remark this answers is not in the document]
Comment (andriidomasov) [reply Packs/Inventa/ReleaseNotes/1_1_0.md]: done

**Comment by idovandijk** [comment]
Hi @andriidomasov. I checked and there's an upcoming feature to contribute to common fields through the UI, but currently you would have to do that through git directly.
Regarding the error you have - the dev team replied that it would be helpful if you could share logs bundle in debug mode, and also configure the integration to run in verbose mode to get the  extra logging.

**Comment by ilappe** [comment]
@andriidomasov 
Thanks for your fixes!
it looks like the release notes you updated in CommonTypes are wrong (it mention `Indicator Fields` instead of Incident Fields ?)
in addition there is a lot of empty files
please remove them if not relevant
thanks!

**Comment by andriidomasov** [comment]
> the dev team replied that it would be helpful if you could share logs bundle in debug mode, and also configure the integration to run in verbose mode to get the extra logging.

Log entries after running command from integration in verbose mode:

2022-06-20 08:43:12.1643 info (Inventa_instance_1_Inventa_inventa-get-sources-piis) cURL:
curl -X GET https://10.192.192.142/pii/api/piis/627d8a2ee45cba5d59f072f4/sources/details -H "Accept: */*" -H "Authorization: Bearer <XX_REPLACED>" --noproxy "*" -k  (source: /builds/gopath/src/github.com/demisto/server/services/automation/dockercoderunner.go:847)
2022-06-20 10:26:44.3856 debug Running Inventa_instance_1_Inventa_inventa-get-datasubject-id. Timeout: 5m0s (source: /builds/gopath/src/github.com/demisto/server/services/automation/dockercoderunner.go:245)
2022-06-20 10:26:44.3856 debug Sending runner request for script [Inventa_instance_1_Inventa_inventa-get-datasubject-id] (source: /builds/gopath/src/github.com/demisto/server/services/automation/dockercoderunner.go:292)
2022-06-20 10:26:44.3857 debug Waiting for runner request for unnested script [Inventa_instance_1_Inventa_inventa-get-datasubject-id], with script timeout [5m0s] (source: /builds/gopath/src/github.com/demisto/server/services/automation/dockercoderunner.go:326)
2022-06-20 10:26:46.0680 error Failed to run script: Inventa_instance_1_Inventa_inventa-get-datasubject-id with image: demisto/python3:3.10.4.30607 [error 'Docker code runner got container error: [Docker code script is in inconsistent state, expected pong but didn't got any response, exit error: Container exit with error. container name: [demistoserver_pyexec-fe754faa-af39-4d1c-8c33-18ba93df5078-demistopython33.10.4.30607--279] error: [exit status 1] stderr: [/usr/local/bin/python: can't find '__main__' module in '/tmp/pyrunner/_script_docker_python_loop.py'
] (2619) (2610)] (2611)'] (source: /builds/gopath/src/github.com/demisto/server/services/automation/runner.go:68)
2022-06-20 11:08:13.1437 debug Running Inventa_instance_1_Inventa_inventa-get-datasubject-id. Timeout: 5m0s (source: /builds/gopath/src/github.com/demisto/server/services/automation/dockercoderunner.go:245)
2022-06-20 11:08:13.1438 debug Sending runner request for script [Inventa_instance_1_Inventa_inventa-get-datasubject-id] (source: /builds/gopath/src/github.com/demisto/server/services/automation/dockercoderunner.go:292)
2022-06-20 11:08:13.1438 debug Waiting for runner request for unnested script [Inventa_instance_1_Inventa_inventa-get-datasubject-id], with script timeout [5m0s] (source: /builds/gopath/src/github.com/demisto/server/services/automation/dockercoderunner.go:326)
2022-06-20 11:08:14.8244 error Failed to run script: Inventa_instance_1_Inventa_inventa-get-datasubject-id with image: demisto/python3:3.10.4.30607 [error 'Docker code runner got container error: [Docker code script is in inconsistent state, expected pong but didn't got any response, exit error: Container exit with error. container name: [demistoserver_pyexec-81139918-a98f-44fa-89b6-4786913b4cba-demistopython33.10.4.30607--282] error: [exit status 1] stderr: [/usr/local/bin/python: can't find '__main__' module in '/tmp/pyrunner/_script_docker_python_loop.py'
] (2619) (2610)] (2611)'] (source: /builds/gopath/src/github.com/demisto/server/services/automation/runner.go:68)
2022-06-20 11:10:36.7763 debug Running Inventa_instance_1_Inventa_inventa-get-datasubject-id. Timeout: 5m0s (source: /builds/gopath/src/github.com/demisto/server/services/automation/dockercoderunner.go:245)
2022-06-20 11:10:36.7764 debug Sending runner request for script [Inventa_instance_1_Inventa_inventa-get-datasubject-id] (source: /builds/gopath/src/github.com/demisto/server/services/automation/dockercoderunner.go:292)
2022-06-20 11:10:36.7764 debug Waiting for runner request for unnested script [Inventa_instance_1_Inventa_inventa-get-datasubject-id], with script timeout [5m0s] (source: /builds/gopath/src/github.com/demisto/server/services/automation/dockercoderunner.go:326)
2022-06-20 11:10:38.3876 error Failed to run script: Inventa_instance_1_Inventa_inventa-get-datasubject-id with image: demisto/python3:3.10.4.30607 [error 'Docker code runner got container error: [Docker code script is in inconsistent state, expected pong but didn't got any response, exit error: Container exit with error. container name: [demistoserver_pyexec-85d5a855-8217-4979-8921-0d14da60177d-demistopython33.10.4.30607--285] error: [exit status 1] stderr: [/usr/local/bin/python: can't find '__main__' module in '/tmp/pyrunner/_script_docker_python_loop.py'
] (2619) (2610)] (2611)'] (source: /builds/gopath/src/github.com/demisto/server/services/automation/runner.go:68)

**Comment by idovandijk** [comment]
@andriidomasov would it be possible to schedule a remote debug session with @wissamg to investigate the error you were getting with the integration?

**Comment by andriidomasov** [comment]
> @andriidomasov would it be possible to schedule a remote debug session with @wissamg to investigate the error you were getting with the integration?

Sure. Please advise suitable time slots for Wed-Fri this week if convenient.

**Comment by wissamg** [comment]
> > @andriidomasov would it be possible to schedule a remote debug session with @wissamg to investigate the error you were getting with the integration?
> 
> Sure. Please advise suitable time slots for Wed-Fri this week if convenient.

Hi All Hi @andriidomasov,
 Yes please, looks like I'm widely free on Wednesday and Thursday.
Just please update me on what TimeZone you are. (the only thing for me on Wednesday I'm working I will busy after 4pm Israel time getting free at 9pm)

**Comment by andriidomasov** [comment]
> > > @andriidomasov would it be possible to schedule a remote debug session with @wissamg to investigate the error you were getting with the integration?
> > 
> > 
> > Sure. Please advise suitable time slots for Wed-Fri this week if convenient.
> 
> Hi All Hi @andriidomasov, Yes please, looks like I'm widely free on Wednesday and Thursday. Just please update me on what TimeZone you are. (the only thing for me on Wednesday I'm working I will busy after 4pm Israel time getting free at 9pm)

I believe we work in the same timezone (Ukraine & Israel). How about 3 P.M. Thursday, 22 June?

**Comment by wissamg** [comment]
> 

Sounds good to me

**Comment by idovandijk** [comment]
So happy to know that the issue was resolved, thank you Wissam.
@andriidomasov in order to proceed we still need to get rid of the empty files in the PR (what are they?) and fix the release-notes as Israel mentioned. Please let us know once you're done making changes.

**Comment by andriidomasov** [comment]
> @andriidomasov
> Thanks for your fixes!
> it looks like the release notes you updated in CommonTypes are wrong (it mention `Indicator Fields` instead of Incident Fields ?)
> in addition there is a lot of empty files
> please remove them if not relevant
> thanks!

Please elaborate more on the empty files. What files are you referring to?

**Comment by ilappe** [comment]
> Please elaborate more on the empty files. What files are you referring to?

Hi @andriidomasov 
see screenshot bellow 

image

image

**Comment by andriidomasov** [comment]
That's very peculiar, as those files are not empty...

Screenshot 2022-06-27 at 14 30 30

**Comment by ilappe** [comment]
@andriidomasov 
it's look like all of those files have change in their permission (100644->100755)
can you please ignore this change and we will check it again ?

**Review by idovandijk — COMMENTED** [review]
I see that @andriidomasov has moved the required fields to Common and also updated the layout which is great.
However I do see that the "Inventa Birthday" and "Inventa Full Name" fields **still exist** in this PR.

@ilappe from my end this PR just needs the 2 fields removed and the interesting permission issue sorted. If there's any way I can help here let me know, otherwise I'm good with this PR other than the 2 points I mentioned.

Thank you all, it's been fun reviewing this work.

**Comment by andriidomasov** [comment]
> @andriidomasov
> it's look like all of those files have change in their permission (100644->100755)
> can you please ignore this change and we will check it again ?

Just to be clear — if I got you right, by ignoring the changes you mean to revert permissions back?

**Comment by ilappe** [comment]
> > @andriidomasov
> > it's look like all of those files have change in their permission (100644->100755)
> > can you please ignore this change and we will check it again ?
> 
> Just to be clear — if I got you right, by ignoring the changes you mean to revert permissions back?

yes - please revert the permission back
in addition please note to the remaining points mentioned in [this comment](https://github.com/demisto/content/pull/19392#pullrequestreview-1020475608)

**Comment by ilappe** [comment]
Hi @andriidomasov 
thanks a lot for your fixes
for now it's looks great 
but please just validate if the 1_1_0.md file in the releaseNotes are correct 
as many files mentioned in it as modified - aren't (for example Packs/Inventa/IncidentFields/incidentfield-Inventa_Report_Reason.json)
image

**Review by ilappe — APPROVED** [review]

**Pull request merged**